oml/functional/metrics.py

```diff
@@ -220,42 +216,209 @@ def calculate_accuracy_on_triplets(embeddings: torch.Tensor, reduce_mean: bool =
         return acc
 
 
+def calc_cmc(gt_tops: torch.Tensor, top_k: Tuple[int, ...]) -> List[torch.Tensor]:
+    """
+    Function to compute Cumulative Matching Characteristics (CMC) for each sample.
+
+    ``cmc@k`` is 1 if ``top_k`` distances contain an element from the same class, and 0 otherwise.
+
+    Args:
+        gt_tops: ground truth of the ``max(top_k)`` distances.
+        top_k: Tuple of ``k`` values to calculate ``cmc@k`` for.
+
+    Returns:
+        List of ``cmc@k`` tensors.
+
+    Example:
+        >>> gt_tops = torch.tensor([[1, 0, 0], [0, 1, 1], [0, 0, 1]], dtype=torch.bool)
+        >>> calc_cmc(gt_tops, (1, 2, 3))
+        [tensor([1., 0., 0.]), tensor([1., 1., 0.]), tensor([1., 1., 1.])]
+    """
+    _check_if_nonempty_integers_and_positive(top_k, "top_k")
+    top_k = _clip_max_with_warning(top_k, gt_tops.shape[1])
+    cmc = []
+    for k in top_k:
+        cmc.append(torch.any(gt_tops[:, :k], dim=1).float())
+    return cmc
+
+
+def calc_precision(gt_tops: torch.Tensor, n_gt: torch.Tensor, top_k: Tuple[int, ...]) -> List[torch.Tensor]:
+    """
+    Function to compute Precision@k for each sample.
+
+    ``precision@k`` is a proportion of the ``top_k`` elements from the same class to total number of the elements
```

Comment:
I would say

``precision@k`` is a fraction of the relevant gallery instances among ``top_k`` instances closest to the corresponding queries.